crates/kubelet/src/secret/mod.rs

```diff
@@ -0,0 +1,112 @@
+//! Resolves image pull secrets
+
+use k8s_openapi::api::core::v1::Secret;
+use kube::api::Api;
+use oci_distribution::secrets::RegistryAuth;
+
+/// Resolves registry authentication from image pull secrets
+pub struct RegistryAuthResolver {
+    kube_client: kube::Client,
+    pod_namespace: String,
+    image_pull_secret_names: Vec<String>,
+}
+
+impl RegistryAuthResolver {
+    /// Creates a resolver for the given pod
+    pub fn new(client: kube::Client, pod: &crate::pod::Pod) -> Self {
+        // TODO: is it safe to capture this stuff or might we need to re-resolve e.g.
```

Comment:
As far as I understand about the pod spec and how we treat a modify, I think it is perfectly safe to capture this